### DISCUSSION TITLE
Error code -133: File couldn't be opened or doesn't exist.

### DISCUSSION THREAD

**Discussion by georgearun** [TeamPyOgg/PyOgg#68] [Q&A] [answered]
```
import pyogg

opus_file = pyogg.OpusFile("C:/voice/renamed.opus")
```

I've written the above code to read the opus file but I'm getting the following error "PyOggError: file couldn't be opened or doesn't exist. Error code  -133" even though the file exists.
Is there a solution to this issue?

**Comment by mattgwwalker** [comment]
One possibility is that the opus file is corrupt. Can you play it with the
likes of VLC or Firefox?

OpusFile doesn’t use the standard python file handling facilities. I’m not
a Windows user, but I’d consider trying swapping out the slashes: / for \.

You could try moving the file to the working directory. That would allow
you to refer to the file as just “renamed.opus”. If it still doesn't work,
then it’s probably a corrupt file. If it does work then it’s likely the
slashes or similar that are causing difficulties.

Please report back to tell us how you go.

Cheers,

Matthew

**Comment by mattgwwalker** [comment]
Also, it would be worthwhile trying the examples out. They should work
without error on your machine. That should give you confidence that PyOgg
is installed correctly.

For even more confidence generation, you could also run the tests. They too
should pass without error.

Both the examples and tests check file reading.

Matthew

**Reply by georgearun** [reply]
Hi @mattgwwalker  I did try both of your suggestions. I could read the example .opus files provided.  Shifting the file location to the same directory and VLC/Media player did not seem to work with the opus files I have.

**Reply by mattgwwalker** [reply]
Just to make sure that I understand you correctly...

You were able to play the .opus files in the examples directory using VLC.  (Great.)

But you were not able to play your own `renamed.opus` file using VLC.  (Not Great.)

If that's the case then the problem lies in the file `renamed.opus`... as in, the file's corrupt.  How did you create/obtain it?  If you are able to share it, I'm happy to double-check that for you.

If I've misunderstood you, please give me more information.

Matthew

**Reply by georgearun** [reply]
The proper source is not known other than it's a voice recording device. I guess you're right the files might be corrupt. 
Sorry, I won't be able to share the files due to data privacy violations. You can close this ticket.

**Comment by mattgwwalker** [comment]
Interesting that it’s a voice recorder. But if VLC can’t play it, then yes,
I would suspect it’s a corrupt file and it’s not a problem with PyOgg.

Sorry I can’t be of more help.

Matthew

**Comment by mattgwwalker** [comment, marked as answer]
If the recording is of significant value to you, you might consider using `opusinfo` (part of [`opustools`](https://github.com/xiph/opus-tools)).

Opusinfo will give you indications of what's wrong with the file.  However I know of no automated tools to fix a corrupt Opus-encoded file.

Cheers,

Matthew
